pkgs/development/octave-modules/symbolic/default.nix

```diff
@@ -0,0 +1,39 @@
+{ buildOctaveLibrary
+, stdenv
+, fetchurl
+, octave
+, gnuplot
+, pythonEnv
+, texinfo
+}:
+
+(buildOctaveLibrary rec {
+  pname = "symbolic";
+  version = "2.9.0";
+
+  src = fetchurl {
+    url = "https://octave.sourceforge.io/download.php?package=${pname}-${version}.tar.gz";
+    sha256 = "1jr3kg9q6r4r4h3hiwq9fli6wsns73rqfzkrg25plha9195c97h8";
+  };
+
+  buildInputs = [ pythonEnv ];
+
+  meta = with stdenv.lib; {
+    homepage = "https://octave.sourceforge.io/${pname}/index.html";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ KarlJoad ];
+    description = "Adds symbolic calculation features to GNU Octave";
+  };
+}).overrideAttrs (oldAttrs: rec {
+  dontUnpack = false;
+  buildPhase = ''
+      substituteInPlace inst/private/defaultpython.m --replace python3 ${pythonEnv}/bin/python
+    '';
+  dontInstall = false;
+  installPhase = ''
+      mkdir -p $out
```

Comment:
Missing hooks... In many other places as well. See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md

Reply:
See https://github.com/NixOS/nixpkgs/pull/108562/commits/17b08c0843c4d51c10e32d45a40e9fb19ea824d7

I am unsure if `symbolic`'s override needs to have those hooks present. 
What I would really prefer to do is completely change `buildPhase` and add an `installPhase` for `symbolic`.

Reply:
I managed to get the override removed by using `makeWrapper` during the environment's building, so now there is no need to decompress the tarball anymore. 

Now, if there is a non-octave package run-time dependency, it should be placed in `propagatedBuildInputs` and a reference to those packages will be made in the wrapper that is generated in `wrapper.nix` and `wrap.sh` now.

If there is a run-time dependency on another octave package, it should be placed elsewhere (right now the name I've decided on is `requiredOctavePackages`), which will install the required package into octave alongside the one that was manually specified.

Reply:
Now, `build-octave-package.nix` has `runHook`s in its `buildPhase` to ensure those are run, although they are not very helpful right now as I don't decompress the tarball, as octave prefers to work with tarballs.